syft/get_source.go

```diff
@@ -0,0 +1,102 @@
+package syft
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/anchore/syft/syft/source"
+)
+
+func GetSource(ctx context.Context, userInput string, getSourceConfig ...GetSourceConfig) (source.Source, error) {
```

Comment:
We should try and be consistent with other top-level functions, such as `CreateSBOM` which takes an optional config with `*ConfigType` instead of `...ConfigType`.

Reply:
I have a soft preference for the varargs approach, but agree about consistency 💯